### ISSUE TITLE
Setting up a path to `ScreenRecorder.ffmpeg_binary =` fails

### ISSUE THREAD

**Issue by Rajagopalan-M** [kapoorlakshya/screen-recorder#84]
I have created my own video recorder folder inside my project and gave that path in my program like

`
ScreenRecorder.ffmpeg_binary = "C:\\A\\RubyAutomation\\VideoRecorder\\ffmpeg.exe"`

But it fails.

**Comment by kapoorlakshya** [comment]
Can you please use the issue template and give me a proper error message or explain the actual behavior? At this point I can only guess what is happening.

**Comment by Rajagopalan-M** [comment]
There is no error message, But your two statement of setting up the the path and opening the recorder is disconnected. Just write the following line 

```
ScreenRecorder.ffmpeg_binary = "C:\\A\\RubyAutomation\\VideoRecorder\\ffmpeg.exe"
@recorder = ScreenRecorder::Window.new(title: 'firefox', output: File.expand_path("Output/Video/filename.mp4"))
```

The first line is no way connected to the second line. No matter wherever you set up the path, it only picks up from what you set in the environmental variable. It's not changing the path while you are setting it.

**Comment by kapoorlakshya** [comment]
Which `screen-recorder` version are you using? And is there a reason why you have the path set in both ENV and `ScreenRecorder.ffmpeg_binary`? Either one should work.

Anyway, this is working for me as expected in v1.3.1:

```
screen-recorder on 84_ffmpeg_bin via 💎 v2.7.0 
➜ bin/console 
irb(main):001:0> ScreenRecorder.ffmpeg_binary
=> "/usr/local/bin/ffmpeg"
irb(main):002:0> ScreenRecorder.ffmpeg_binary = "./test_bin/ffmpeg" # This bin is not in ENV
irb(main):003:0> ScreenRecorder.ffmpeg_binary
=> "./test_bin/ffmpeg"
irb(main):004:0> @recorder = ScreenRecorder::Desktop.new(output: 'test.mkv')
irb(main):005:0> @recorder.start
=> #<ChildProcess::Unix::PosixSpawnProcess:0x00007fe8ef058e28 @args=["sh", "-c", "./coverage/ffmpeg -y -f avfoundation -pix_fmt uyvy422 -i 1  -pix_fmt yuv420p -framerate 15.0 -vf \"scale=trunc(iw/2)*2:trunc(ih/2)*2\" test.mkv"], @started=true, @exit_code=nil, @io=#<ChildProcess::Unix::IO:0x00007fe8ef058d88 @stderr=#<File:ffmpeg.log>, @stdout=#<File:ffmpeg.log>, @stdin=#<IO:fd 11>>, @cwd=nil, @detach=false, @duplex=true, @leader=false, @environment={}, @pid=62499>
irb(main):006:0> @recorder.stop
=> #<FFMPEG::Movie:0x00007fe90f32c6c0 @path="test.mkv", @metadata={:streams=>[{:index=>0, :codec_name=>"h264", :codec_long_name=>"H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10", :profile=>"High", :codec_type=>"video", :codec_time_base=>"1/60", :codec_tag_string=>"[0][0][0][0]", :codec_tag=>"0x0000", :width=>3584, :height=>2240, :coded_width=>3584, :coded_height=>2240, :has_b_frames=>2, :pix_fmt=>"yuv420p", :level=>62, :chroma_location=>"left", :field_order=>"progressive", :refs=>1, :is_avc=>"true", :nal_length_size=>"4", :r_frame_rate=>"30/1", :avg_frame_rate=>"30/1", :time_base=>"1/1000", :start_pts=>0, :start_time=>"0.000000", :bits_per_raw_sample=>"8", :disposition=>{:default=>1, :dub=>0, :original=>0, :comment=>0, :lyrics=>0, :karaoke=>0, :forced=>0, :hearing_impaired=>0, :visual_impaired=>0, :clean_effects=>0, :attached_pic=>0, :timed_thumbnails=>0}, :tags=>{:ENCODER=>"Lavc58.54.100 libx264", :DURATION=>"00:00:03.433000000"}}], :format=>{:filename=>"test.mkv", :nb_streams=>1, :nb_programs=>0, :format_name=>"matroska,webm", :format_long_name=>"Matroska / WebM", :start_time=>"0.000000", :duration=>"3.433000", :size=>"302144", :bit_rate=>"704093", :probe_score=>100, :tags=>{:ENCODER=>"Lavf58.29.100"}}}, @container="matroska,webm", @duration=3.433, @time=0.0, @format_tags={:ENCODER=>"Lavf58.29.100"}, @creation_time=nil, @bitrate=704093, @video_codec="h264", @colorspace="yuv420p", @width=3584, @height=2240, @video_bitrate=0, @sar=nil, @dar=nil, @frame_rate=(30/1), @video_stream="h264 (High) ([0][0][0][0] / 0x0000), yuv420p, 3584x2240 [SAR  DAR ]", @rotation=nil, @audio_streams=[]>
irb(main):007:0> 
```

If you look at the code below, you'll notice that the gem is designed to use whatever binary location `ScreenRecorder.ffmpeg_binary` returns. And it will only look in ENV if you don't set the binary path in your project.

https://github.com/kapoorlakshya/screen-recorder/blob/42148992868e826eb52c46580f5a3f350d8d6bf4/lib/screen-recorder/common.rb#L115

**Comment by Rajagopalan-M** [comment]
Setting up here `ScreenRecorder.ffmpeg_binary =` here returned correctly from `ScreenRecorder.ffmpeg_binary` But the actual recorder which is used from what's set on the environmental variable. Now I have removed the path from the environmental variable and you could see the following error is thrown
```

INFO: Could not find files for the given pattern(s).


ScreenRecorder::Errors::DependencyNotFound: ffmpeg binary not found.

  0) StingRay(AGCS) Scenario22-VerifyEndorsementPre AGCSQuoteCreation_Test1
     Failure/Error: raise Errors::DependencyNotFound, 'ffmpeg binary not found.' unless ffmpeg_exists?

     ScreenRecorder::Errors::DependencyNotFound:
       ffmpeg binary not found.
     # SequencialRun.rb:214:in `block (5 levels) in <top (required)>'

```

**Comment by kapoorlakshya** [comment]
Hmm... thanks for that information. Looks like a valid bug.

I'll look into it this tomorrow.

**Comment by Rajagopalan-M** [comment]
Thank you very much.

**Comment by Rajagopalan-M** [comment]
Oh I think you seems to have fixed this, thanks. Let you know once after I check it.

**Comment by kapoorlakshya** [comment]
I have released v1.4.0 with fix for this and #85. You can read the changelog [here](https://github.com/kapoorlakshya/screen-recorder/blob/master/CHANGELOG.md).

Let me know if you run into any other issues.
